payu/manifest.py

```diff
@@ -286,9 +298,10 @@ def __init__(self, config, reproduce):
         # Set flag to auto-scan input directories
         self.scaninputs = self.manifest_config.get('scaninputs', True)
 
-        if self.reproduce['input'] and self.scaninputs:
-            print("scaninputs set to False when reproduce input is True")
-            self.scaninputs = False
+        # Ensure configured input files are discovered when reproduce is True
+        if self.reproduce['input'] and not self.scaninputs:
+            print("scaninputs set to True when reproduce input is True")
+            self.scaninputs = True
```

Comment:
Ok I had a go at removing scaninputs in bfb2bb0. It does simplify the logic quite a bit. The documentation for scanInputs was: 
```     
 If a manifest file is complete and it is desirable to not add spurious
      files to the manifest but allow existing files to change, setting this
      option to *False* would allow that behaviour.
````
I don't know what it was referring to about spurious files, so I don't know the reason of why it was an option initially. If decide not to remove it, can revert back to an earlier commit.

Reply:
Yeah it was always a bit vague about the use case. Essentially people had a habit of pointing at a directory which had a load of stuff in it, including random files that weren't used by the model. So pruning the manifest could be a way of narrowing down the files that are being tracked. It was always a bad solution. The real solution is to keep things clean, and more lately just reference the files being used directly in `config.yaml`.